sale_order_product_availability_inline/models/product_product.py

```diff
@@ -11,7 +11,7 @@ def name_get(self):
         if self.env.context.get("so_product_stock_inline"):
             res = super().name_get()
             self = self.with_context(warehouse=self.env.context.get("warehouse"))
-            availability = {r.id: [r.free_qty, r.uom_id.name] for r in self}
+            availability = {r.id: [r.free_qty, r.uom_id.name_get()[0][1]] for r in self}
```

Comment:
Why this change? In any case, why not using `display_name`. A tip from the code:

https://github.com/odoo/odoo/blob/c79bda62edd8316c2a5754d8aa8838f8aa4caee0/odoo/models.py#L1821